scripts/install_linux.sh

```diff
@@ -58,25 +62,25 @@ fi
 
 echo "Downloading tfsec $version"
 
-curl --fail --silent -L -o "${download_path}/${remote_filename}" "https://github.com/aquasecurity/tfsec/releases/download/${version}/${remote_filename}"
-tfsec_dl_status=$?
-if [ $tfsec_dl_status -ne 0 ]; then
-  echo "Failed to download ${remote_filename}"
-  exit $tfsec_dl_status
-fi
-echo "Downloaded successfully"
-
-echo "Validate checksum"
-curl --fail --silent -L -o "${download_path}/${checksum_file}" "https://github.com/aquasecurity/tfsec/releases/download/${version}/${checksum_file}"
-checksum_dl_val=$?
-if [ $checksum_dl_val -ne 0 ]; then
-  echo "Failed to download checksum file ${checksum_file}"
-  exit $checksum_dl_val
-fi
+download_file() {
+  echo "Downloading $3..."
+  curl --fail --silent -L -o "$1/$3" "https://github.com/aquasecurity/tfsec/releases/download/$2/$3"
```

Comment:
recommendation:

```bash
download_file() {
   local download_path=${1:?Download path no supplied}   
   local version=${2:?No version supplied}
   local file=${3:?File to download not supplied}
   curl --fail-with-body --fail --silent -L -o "${download_path}/${file}" "https://github.com/aquasecurity/tfsec/releases/download/${version}/${file}"
}
```
This has a few advantages:
1. it asserts that params were passed or the script returns an error
2. make it obvious which params go where and represents what by naming them
3. On failure, outputs the body if there was an error (optional, but good for debug)

Reply:
@chb0github thank you for the detailed review. Will edit and resubmit. Thank you!